diff -ur qgis-3.32.2.orig/CMakeLists.txt qgis-3.32.2/CMakeLists.txt
--- qgis-3.32.2.orig/CMakeLists.txt	2023-08-28 05:34:18.593766584 +0200
+++ qgis-3.32.2/CMakeLists.txt	2023-08-28 05:34:35.843639201 +0200
@@ -363,7 +363,12 @@
     message (SEND_ERROR "sqlite3 dependency was not found!")
   endif()
 
-  find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
+  # try upstream CMake config first
+  find_package(Protobuf CONFIG)
+  if(NOT Protobuf_FOUND)
+    # fall back to CMake's protobuf module
+    find_package(Protobuf REQUIRED)   # for decoding of vector tiles in MVT format
+  endif()
   message(STATUS "Found Protobuf: ${Protobuf_LIBRARIES}")
   if (NOT Protobuf_PROTOC_EXECUTABLE)
     message (SEND_ERROR "Protobuf library's 'protoc' tool was not found!")
diff -ur qgis-3.32.2.orig/src/core/CMakeLists.txt qgis-3.32.2/src/core/CMakeLists.txt
--- qgis-3.32.2.orig/src/core/CMakeLists.txt	2023-08-28 05:34:18.523767101 +0200
+++ qgis-3.32.2/src/core/CMakeLists.txt	2023-08-28 05:35:18.803321963 +0200
@@ -2375,7 +2375,7 @@
   ${EXPAT_LIBRARY}
   ${SQLITE3_LIBRARY}
   ${LIBZIP_LIBRARY}
-  ${Protobuf_LITE_LIBRARY}
+  protobuf::libprotobuf-lite
   ${ZLIB_LIBRARIES}
   ${EXIV2_LIBRARY}
 )
