diff -ur a/common/profile-sync-daemon.in b/common/profile-sync-daemon.in
--- a/common/profile-sync-daemon.in	2015-07-29 22:57:24.000000000 +0200
+++ b/common/profile-sync-daemon.in	2016-09-28 17:26:35.897211630 +0200
@@ -581,7 +581,7 @@
 				# check that the LAST DIRECTORY in the full path is unique
 				unique_count=$(echo ${DIRArr[@]##*/} |
 				sed 's/ /\n/g' | sort -u | wc -l)
-				if [[ ${#DIRArr[@]##*/} -eq $unique_count ]]; then
+				if [[ ${#DIRArr[@]} -eq $unique_count ]]; then
 					# no problems so do nothing
 					return
 				else
@@ -603,7 +603,7 @@
 				# check that the LAST DIRECTORY in the full path is unique
 				unique_count=$(echo ${DIRArr[@]##*/} |
 				sed 's/ /\n/g' | sort -u | wc -l)
-				if [[ ${#DIRArr[@]##*/} -eq $unique_count ]]; then
+				if [[ ${#DIRArr[@]} -eq $unique_count ]]; then
 					# no problems so do nothing
 					return
 				else
