diff -ur flann-1.9.1.orig/src/cpp/CMakeLists.txt flann-1.9.1/src/cpp/CMakeLists.txt
--- flann-1.9.1.orig/src/cpp/CMakeLists.txt	2018-04-17 20:42:10.257351180 +0200
+++ flann-1.9.1/src/cpp/CMakeLists.txt	2018-04-17 20:49:46.981239435 +0200
@@ -28,25 +28,11 @@
     set_property(TARGET flann_cuda_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-    add_library(flann_cpp SHARED "")
-    set_target_properties(flann_cpp PROPERTIES LINKER_LANGUAGE CXX)
-    target_link_libraries(flann_cpp -Wl,-whole-archive flann_cpp_s -Wl,-no-whole-archive)
-
-    if (BUILD_CUDA_LIB)
-	    cuda_add_library(flann_cuda SHARED "")
-        set_target_properties(flann_cuda PROPERTIES LINKER_LANGUAGE CXX)
-        target_link_libraries(flann_cuda -Wl,-whole-archive flann_cuda_s -Wl,-no-whole-archive)
-        set_property(TARGET flann_cpp_s PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
-    # 	target_link_libraries(flann_cuda cudpp_x86_64)
-    endif()
-else()
     add_library(flann_cpp SHARED ${CPP_SOURCES})
     if (BUILD_CUDA_LIB)
 		cuda_add_library(flann_cuda SHARED ${CPP_SOURCES})
         set_property(TARGET flann_cpp PROPERTY COMPILE_DEFINITIONS FLANN_USE_CUDA)
     endif()
-endif()
 
 set_target_properties(flann_cpp PROPERTIES
    VERSION ${FLANN_VERSION}
@@ -82,17 +68,11 @@
     endif()
     set_property(TARGET flann_s PROPERTY COMPILE_DEFINITIONS FLANN_STATIC)
 
-    if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_COMPILER_IS_GNUCC)
-        add_library(flann SHARED "")
-        set_target_properties(flann PROPERTIES LINKER_LANGUAGE CXX)
-        target_link_libraries(flann -Wl,-whole-archive flann_s -Wl,-no-whole-archive)
-    else()
         add_library(flann SHARED ${C_SOURCES})
 
         if(MINGW AND OPENMP_FOUND)
           target_link_libraries(flann gomp)
         endif()
-    endif()
 
     set_target_properties(flann PROPERTIES
        VERSION ${FLANN_VERSION}
