reverted:
--- b/cmake/systemd.cmake
+++ a/cmake/systemd.cmake
@@ -76,11 +76,9 @@
         UNSET(HAVE_SYSTEMD_SD_NOTIFYF)
         MESSAGE(STATUS "Systemd features not enabled")
         IF(WITH_SYSTEMD STREQUAL "yes")
+          MESSAGE(FATAL_ERROR "Requested WITH_SYSTEMD=YES however no dependencies installed/found")
-          MESSAGE(FATAL_ERROR "Requested WITH_SYSTEMD=yes however no dependencies installed/found")
         ENDIF()
       ENDIF()
-    ELSEIF(NOT WITH_SYSTEMD STREQUAL "no")
-      MESSAGE(FATAL_ERROR "Invalid value for WITH_SYSTEMD. Must be 'yes', 'no', or 'auto'.")
     ENDIF()
   ENDIF()
 ENDMACRO()
