diff -ur vmware-host-modules-efc76a69acebb66409af5f267c51ccc027e057e6.orig/vmnet-only/bridge.c vmware-host-modules-efc76a69acebb66409af5f267c51ccc027e057e6/vmnet-only/bridge.c
--- vmware-host-modules-efc76a69acebb66409af5f267c51ccc027e057e6.orig/vmnet-only/bridge.c	2023-02-17 09:50:22.000000000 +0100
+++ vmware-host-modules-efc76a69acebb66409af5f267c51ccc027e057e6/vmnet-only/bridge.c	2023-08-12 15:50:11.915654613 +0200
@@ -45,6 +45,13 @@
 #include <net/tcp.h>
 #include <net/ipv6.h>
 
+/* This fixes moving skb_gso_segment(skb, 0); to net/gso.h from kernel 6.4.10
+*  and later.
+*/
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 4, 10)
+#  include <net/gso.h>
+#endif
+
 #ifdef CONFIG_NET_RADIO
 #   include <linux/wireless.h>
 #endif
