diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 8b289ec..b132fb0 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -41,6 +41,9 @@ endif ()
 # Set the install prefix
 target_compile_definitions(sdrpp_core PUBLIC INSTALL_PREFIX="${CMAKE_INSTALL_PREFIX}")
 
+# Set the plugin prefix
+target_compile_definitions(sdrpp_core PUBLIC PLUGINS_PREFIX="${CMAKE_INSTALL_FULL_LIBDIR}")
+
 # Include core headers
 target_include_directories(sdrpp_core PUBLIC "src/")
 target_include_directories(sdrpp_core PUBLIC "src/imgui")
diff --git a/core/src/core.cpp b/core/src/core.cpp
index b54cf31..591fdec 100644
--- a/core/src/core.cpp
+++ b/core/src/core.cpp
@@ -32,6 +32,14 @@
 #endif
 #endif
 
+#ifndef PLUGINS_PREFIX
+#ifdef __APPLE__
+#define PLUGINS_PREFIX "/usr/local/lib"
+#else
+#define PLUGINS_PREFIX "/usr/lib"
+#endif
+#endif
+
 namespace core {
     ConfigManager configManager;
     ModuleManager moduleManager;
@@ -250,7 +258,7 @@ int sdrpp_main(int argc, char* argv[]) {
     defConfig["modulesDirectory"] = root + "/modules";
     defConfig["resourcesDirectory"] = root + "/res";
 #else
-    defConfig["modulesDirectory"] = INSTALL_PREFIX "/lib/sdrpp/plugins";
+    defConfig["modulesDirectory"] = PLUGINS_PREFIX "/sdrpp/plugins";
     defConfig["resourcesDirectory"] = INSTALL_PREFIX "/share/sdrpp";
 #endif
 
