diff -ur swreceiver.orig/includes/scopes-qwt6/scope.h swreceiver/includes/scopes-qwt6/scope.h
--- swreceiver.orig/includes/scopes-qwt6/scope.h	2022-07-28 15:20:31.648114746 +0200
+++ swreceiver/includes/scopes-qwt6/scope.h	2022-07-28 15:21:58.128452010 +0200
@@ -41,6 +41,7 @@
 #include	<qwt_plot_layout.h>
 #include	<qwt_scale_widget.h>
 #include	<QBrush>
+#include	<QPen>
 #include	<QTimer>
 #include	<stdint.h>
 #include	"spectrogramdata.h"
diff -ur swreceiver.orig/includes/scopes-qwt6/spectrogramdata.h swreceiver/includes/scopes-qwt6/spectrogramdata.h
--- swreceiver.orig/includes/scopes-qwt6/spectrogramdata.h	2022-07-28 15:20:31.648114746 +0200
+++ swreceiver/includes/scopes-qwt6/spectrogramdata.h	2022-07-28 15:21:15.144290298 +0200
@@ -29,9 +29,17 @@
 
 #include	"swradio-constants.h"
 #include	<qwt_interval.h>
-#include	<qwt_raster_data.h>
+#if defined QWT_VERSION && ((QWT_VERSION >> 8) < 0x0602)
+# include	<qwt_raster_data.h>
+#else
+# include	<qwt_matrix_raster_data.h>
+#endif
 
+#if defined QWT_VERSION && ((QWT_VERSION >> 8) < 0x0602)
 class	SpectrogramData: public QwtRasterData {
+#else
+class	SpectrogramData: public QwtMatrixRasterData {
+#endif
 public:
 	double	*data;		// pointer to actual data
 	int	left;		// index of left most element in raster
@@ -43,7 +51,11 @@
 
 	SpectrogramData (double *data, int left, int width, int height,
 	                 int datawidth, double max):
+#if defined QWT_VERSION && ((QWT_VERSION >> 8) < 0x0602)
         QwtRasterData () {
+#else
+        QwtMatrixRasterData () {
+#endif
 	this	-> data		= data;
 	this	-> left		= left;
 	this	-> width	= width;
