diff -ur valgrind-3.15.0.orig/coregrind/m_redir.c valgrind-3.15.0/coregrind/m_redir.c
--- valgrind-3.15.0.orig/coregrind/m_redir.c	2019-04-13 12:56:21.000000000 +0200
+++ valgrind-3.15.0/coregrind/m_redir.c	2020-05-16 02:24:09.300965843 +0200
@@ -1342,9 +1342,6 @@
       add_hardwired_spec(
          "ld-linux.so.2", "index",
          (Addr)&VG_(x86_linux_REDIR_FOR_index), mandatory);
-      add_hardwired_spec(
-         "ld-linux.so.2", "strlen",
-         (Addr)&VG_(x86_linux_REDIR_FOR_strlen), mandatory);
    }
 
 #  elif defined(VGP_amd64_linux)
@@ -1369,17 +1366,6 @@
       add_hardwired_spec(
          "ld-linux-x86-64.so.2", "index",
          (Addr)&VG_(amd64_linux_REDIR_FOR_index), NULL);
-      add_hardwired_spec(
-         "ld-linux-x86-64.so.2", "strlen",
-         (Addr)&VG_(amd64_linux_REDIR_FOR_strlen),
-#        ifndef GLIBC_MANDATORY_STRLEN_REDIRECT
-         NULL
-#        else
-         /* for glibc-2.10 and later, this is mandatory - can't sanely
-            continue without it */
-         complain_about_stripped_glibc_ldso
-#        endif
-      );   
    }
 
 #  elif defined(VGP_ppc32_linux)
@@ -1387,12 +1373,6 @@
       the start, otherwise ld.so makes a lot of noise. */
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
 
-      /* this is mandatory - can't sanely continue without it */
-      add_hardwired_spec(
-         "ld.so.1", "strlen",
-         (Addr)&VG_(ppc32_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );   
       add_hardwired_spec(
          "ld.so.1", "strcmp",
          (Addr)&VG_(ppc32_linux_REDIR_FOR_strcmp),
@@ -1412,13 +1392,6 @@
       the start, otherwise ld.so makes a lot of noise. */
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
 
-      /* this is mandatory - can't sanely continue without it */
-      add_hardwired_spec(
-         "ld64.so.1", "strlen",
-         (Addr)VG_(fnptr_to_fnentry)( &VG_(ppc64_linux_REDIR_FOR_strlen) ),
-         complain_about_stripped_glibc_ldso
-      );
-
       add_hardwired_spec(
          "ld64.so.1", "index",
          (Addr)VG_(fnptr_to_fnentry)( &VG_(ppc64_linux_REDIR_FOR_strchr) ),
@@ -1433,13 +1406,6 @@
     */
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
 
-      /* this is mandatory - can't sanely continue without it */
-      add_hardwired_spec(
-         "ld64.so.2", "strlen",
-         (Addr)&VG_(ppc64_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
-
       add_hardwired_spec(
          "ld64.so.2", "index",
          (Addr)&VG_(ppc64_linux_REDIR_FOR_strchr),
@@ -1454,17 +1420,6 @@
       distros, ld.so's soname is ld-linux.so.3, but Ubuntu 14.04 on
       Odroid uses ld-linux-armhf.so.3 for some reason. */
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
-      /* strlen */
-      add_hardwired_spec(
-         "ld-linux.so.3", "strlen",
-         (Addr)&VG_(arm_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
-      add_hardwired_spec(
-         "ld-linux-armhf.so.3", "strlen",
-         (Addr)&VG_(arm_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
       /* memcpy */
       add_hardwired_spec(
          "ld-linux.so.3", "memcpy",
@@ -1505,11 +1460,6 @@
       the start, otherwise ld.so makes a lot of noise. */
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
       add_hardwired_spec(
-         "ld-linux-aarch64.so.1", "strlen",
-         (Addr)&VG_(arm64_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
-      add_hardwired_spec(
          "ld-linux-aarch64.so.1", "index",
          (Addr)&VG_(arm64_linux_REDIR_FOR_index),
          NULL 
@@ -1519,13 +1469,6 @@
          (Addr)&VG_(arm64_linux_REDIR_FOR_strcmp),
          NULL 
       );
-#     if defined(VGPV_arm64_linux_android)
-      add_hardwired_spec(
-         "NONE", "__dl_strlen", // in /system/bin/linker64
-         (Addr)&VG_(arm64_linux_REDIR_FOR_strlen),
-         NULL
-      );
-#     endif
    }
 
 #  elif defined(VGP_x86_darwin)
@@ -1534,8 +1477,6 @@
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
       add_hardwired_spec("dyld", "strcmp",
                          (Addr)&VG_(x86_darwin_REDIR_FOR_strcmp), NULL);
-      add_hardwired_spec("dyld", "strlen",
-                         (Addr)&VG_(x86_darwin_REDIR_FOR_strlen), NULL);
       add_hardwired_spec("dyld", "strcat",
                          (Addr)&VG_(x86_darwin_REDIR_FOR_strcat), NULL);
       add_hardwired_spec("dyld", "strcpy",
@@ -1550,8 +1491,6 @@
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
       add_hardwired_spec("dyld", "strcmp",
                          (Addr)&VG_(amd64_darwin_REDIR_FOR_strcmp), NULL);
-      add_hardwired_spec("dyld", "strlen",
-                         (Addr)&VG_(amd64_darwin_REDIR_FOR_strlen), NULL);
       add_hardwired_spec("dyld", "strcat",
                          (Addr)&VG_(amd64_darwin_REDIR_FOR_strcat), NULL);
       add_hardwired_spec("dyld", "strcpy",
@@ -1579,32 +1518,14 @@
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
       /* this is mandatory - can't sanely continue without it */
       add_hardwired_spec(
-         "ld.so.1", "strlen",
-         (Addr)&VG_(mips32_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
-      add_hardwired_spec(
          "ld.so.1", "index",
          (Addr)&VG_(mips32_linux_REDIR_FOR_index),
          complain_about_stripped_glibc_ldso
       );
-#     if defined(VGPV_mips32_linux_android)
-      add_hardwired_spec(
-         "NONE", "__dl_strlen",
-         (Addr)&VG_(mips32_linux_REDIR_FOR_strlen),
-         NULL
-      );
-#     endif
    }
 
 #  elif defined(VGP_mips64_linux)
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
-      /* this is mandatory - can't sanely continue without it */
-      add_hardwired_spec(
-         "ld.so.1", "strlen",
-         (Addr)&VG_(mips64_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
       add_hardwired_spec(
          "ld.so.1", "index",
          (Addr)&VG_(mips64_linux_REDIR_FOR_index),
@@ -1612,11 +1533,6 @@
       );
 #     if defined(VGABI_64)
       add_hardwired_spec(
-         "ld-linux-mipsn8.so.1", "strlen",
-         (Addr)&VG_(mips64_linux_REDIR_FOR_strlen),
-         complain_about_stripped_glibc_ldso
-      );
-      add_hardwired_spec(
          "ld-linux-mipsn8.so.1", "index",
          (Addr)&VG_(mips64_linux_REDIR_FOR_index),
          complain_about_stripped_glibc_ldso
@@ -1639,10 +1555,6 @@
       add_hardwired_spec("/lib/ld.so.1", "strcmp",
                          (Addr)&VG_(x86_solaris_REDIR_FOR_strcmp), NULL);
    }
-   if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
-      add_hardwired_spec("/lib/ld.so.1", "strlen",
-                         (Addr)&VG_(x86_solaris_REDIR_FOR_strlen), NULL);
-   }
 
 #  elif defined(VGP_amd64_solaris)
    if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
@@ -1661,10 +1573,6 @@
       add_hardwired_spec("/lib/amd64/ld.so.1", "strcat",
                          (Addr)&VG_(amd64_solaris_REDIR_FOR_strcat), NULL);
    }
-   if (0==VG_(strcmp)("Memcheck", VG_(details).name)) {
-      add_hardwired_spec("/lib/amd64/ld.so.1", "strlen",
-                         (Addr)&VG_(amd64_solaris_REDIR_FOR_strlen), NULL);
-   }
 
 #  else
 #    error Unknown platform
