diff -ur portaudio.orig/src/hostapi/alsa/pa_linux_alsa.c portaudio/src/hostapi/alsa/pa_linux_alsa.c
--- portaudio.orig/src/hostapi/alsa/pa_linux_alsa.c	2021-04-01 02:35:42.000000000 +0200
+++ portaudio/src/hostapi/alsa/pa_linux_alsa.c	2022-07-28 14:22:03.913391214 +0200
@@ -839,7 +839,7 @@
     PaError result = paNoError;
     snd_pcm_hw_params_t *hwParams;
     snd_pcm_uframes_t alsaBufferFrames, alsaPeriodFrames;
-    unsigned int minChans, maxChans;
+    unsigned int minChans = 0, maxChans = 0;
     int* minChannels, * maxChannels;
     double * defaultLowLatency, * defaultHighLatency, * defaultSampleRate =
         &devInfo->baseDeviceInfo.defaultSampleRate;
@@ -897,7 +897,7 @@
 
     ENSURE_( alsa_snd_pcm_hw_params_get_channels_min( hwParams, &minChans ), paUnanticipatedHostError );
     ENSURE_( alsa_snd_pcm_hw_params_get_channels_max( hwParams, &maxChans ), paUnanticipatedHostError );
-    assert( maxChans <= INT_MAX );
+    assert( maxChans <= UINT_MAX );
     assert( maxChans > 0 );    /* Weird linking issue could cause wrong version of ALSA symbols to be called,
                                    resulting in zeroed values */
 
