diff -ur a/common.mak b/common.mak
--- a/common.mak	2015-04-06 15:41:32.071101688 +0200
+++ b/common.mak	2015-04-06 15:45:13.639344653 +0200
@@ -2,6 +2,7 @@
 
 PREFIX ?= /usr/local
 OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
+LIBDIR ?= lib
 ENABLE_CONVOLUTION ?= no
 FONTFILE?=/usr/share/fonts/truetype/ttf-bitstream-vera/VeraBd.ttf
 VERSION?=$(shell git describe --tags HEAD | sed 's/-g.*$$//;s/^v//' || true)
@@ -17,14 +18,16 @@
 bindir = $(PREFIX)/bin
 sharedir = $(PREFIX)/share/setBfree
 lv2dir = $(PREFIX)/lib/lv2
+lv2dir = $(PREFIX)/$(LIBDIR)/lv2
 
-CFLAGS ?= $(OPTIMIZATIONS) -Wall
+CFLAGS ?= -Wall
 ifeq ($(XWIN),)
 override CFLAGS += -fPIC
 endif
 override CFLAGS += -DVERSION="\"$(VERSION)\""
+override CFLAGS += -fpermissive
 
-CXXFLAGS = $(OPTIMIZATIONS) -Wall
+CXXFLAGS = -Wall
 
 # check for LV2
 LV2AVAIL=$(shell pkg-config --exists lv2 && echo yes)
