--- a/common.mak	2015-11-07 15:09:19.567306775 +0100
+++ b/common.mak	2015-11-07 16:06:12.690968367 +0100
@@ -2,6 +2,7 @@
 
 PREFIX ?= /usr/local
 OPTIMIZATIONS ?= -msse -msse2 -mfpmath=sse -ffast-math -fomit-frame-pointer -O3 -fno-finite-math-only
+LIBDIR ?= lib
 ENABLE_CONVOLUTION ?= no
 FONTFILE?=/usr/share/fonts/truetype/ttf-bitstream-vera/VeraBd.ttf
 VERSION?=$(shell git describe --tags HEAD 2>/dev/null | sed 's/-g.*$$//;s/^v//' || true)
@@ -19,17 +20,18 @@
 
 bindir = $(PREFIX)/bin
 sharedir = $(PREFIX)/share/setBfree
-lv2dir = $(PREFIX)/lib/lv2
+lv2dir = $(PREFIX)/$(LIBDIR)/lv2
 
-CFLAGS ?= $(OPTIMIZATIONS) -Wall
+CFLAGS ?= -Wall
 ifeq ($(XWIN),)
 override CFLAGS += -fPIC
 endif
 override CFLAGS += -DVERSION="\"$(VERSION)\""
+override CFLAGS += -fpermissive
 
 STRIP    ?= strip
 
-CXXFLAGS ?= $(OPTIMIZATIONS) -Wall
+CXXFLAGS = -Wall
 GLUICFLAGS=-I. -I.. -Wno-unused-function
 STRIPFLAGS=-s
 
