diff -ur liquid-dsp-1.3.2.orig/makefile.in liquid-dsp-1.3.2/makefile.in
--- liquid-dsp-1.3.2.orig/makefile.in	2019-07-28 17:52:30.000000000 +0200
+++ liquid-dsp-1.3.2/makefile.in	2019-11-08 17:08:28.935592151 +0100
@@ -1218,9 +1218,9 @@
 install: all
 	@echo "installing..."
 	@echo ""
-	mkdir -p $(DESTDIR)$(exec_prefix)$(libdir)
+	mkdir -p $(DESTDIR)$(libdir)
 	mkdir -p $(DESTDIR)$(prefix)/include/liquid
-	install -m 644 -p ${ARCHIVE_LIB} ${SHARED_LIB} $(DESTDIR)$(exec_prefix)$(libdir)
+	install -m 644 -p ${ARCHIVE_LIB} ${SHARED_LIB} $(DESTDIR)$(libdir)
 	install -m 644 -p $(addprefix include/,$(headers_install)) $(DESTDIR)$(prefix)/include/liquid
 	@echo ""
 	@echo "---------------------------------------------------------"
