diff -ur MLV-App-QTv1.14.orig/platform/qt/MLVApp.pro MLV-App-QTv1.14/platform/qt/MLVApp.pro
--- MLV-App-QTv1.14.orig/platform/qt/MLVApp.pro	2022-07-01 20:59:17.000000000 +0200
+++ MLV-App-QTv1.14/platform/qt/MLVApp.pro	2023-06-15 13:01:47.570258482 +0200
@@ -106,8 +106,8 @@
 
 # Linux
 linux-g++*{
-    QMAKE_CFLAGS += -O3 -fopenmp -msse4.1 -mssse3 -msse3 -msse2 -msse -std=c99 -ftree-vectorize
-    QMAKE_CXXFLAGS += -fopenmp -std=c++11 -ftree-vectorize
+    QMAKE_CFLAGS += -fopenmp -std=c99
+    QMAKE_CXXFLAGS += -fopenmp
     LIBS += -lgomp
 }
 
