diff -ur transmission-4.0.0-beta.2+rbceb368f1b.orig/libtransmission/torrent-magnet.cc transmission-4.0.0-beta.2+rbceb368f1b/libtransmission/torrent-magnet.cc
--- transmission-4.0.0-beta.2+rbceb368f1b.orig/libtransmission/torrent-magnet.cc	2022-12-02 01:09:33.000000000 +0100
+++ transmission-4.0.0-beta.2+rbceb368f1b/libtransmission/torrent-magnet.cc	2022-12-28 18:13:03.491481786 +0100
@@ -302,7 +302,7 @@
     {
         delete tor->incompleteMetadata;
         tor->incompleteMetadata = nullptr;
-        tor->isStopping = true;
+        tor->isStopping = tor->session->shouldPauseAddedTorrents();
         tor->magnetVerify = true;
         if (tor->session->shouldPauseAddedTorrents())
         {
