diff -ur dotherside-0.9.0.orig/test/test_dotherside.cpp dotherside-0.9.0/test/test_dotherside.cpp
--- dotherside-0.9.0.orig/test/test_dotherside.cpp	2022-08-11 22:24:16.000000000 +0200
+++ dotherside-0.9.0/test/test_dotherside.cpp	2022-09-18 18:59:22.413149086 +0200
@@ -279,6 +279,7 @@
 
     void testLoadUrl()
     {
+        QSKIP("this one wants GLX");
         void *url = dos_qurl_create("qrc:///main.qml", QUrl::TolerantMode);
         QVERIFY(url != nullptr);
         dos_qqmlapplicationengine_load_url(m_engine, url);
@@ -290,6 +291,7 @@
 
     void testLoadData()
     {
+        QSKIP("this one wants GLX");
         dos_qqmlapplicationengine_load_data(m_engine, "import QtQuick 2.12; import QtQuick.Controls 2.12; ApplicationWindow { objectName: \"testWindow\"}");
         QCOMPARE(engine()->rootObjects().size(), 1);
         QCOMPARE(engine()->rootObjects().front()->objectName(), QString::fromLocal8Bit("testWindow"));
@@ -769,12 +771,14 @@
 private slots:
     void testCreate()
     {
+        QSKIP("this one wants GLX");
         VoidPointer view(dos_qquickview_create(), &dos_qquickview_delete);
         QVERIFY(view.get());
     }
 
     void testSourceAndSetSource()
     {
+        QSKIP("this one wants GLX");
         std::string testUrl = "qrc:/testQQuickView.qml";
         VoidPointer view(dos_qquickview_create(), &dos_qquickview_delete);
         VoidPointer url(dos_qurl_create(testUrl.c_str(), QUrl::StrictMode), &dos_qurl_delete);
