diff -urN dotherside-0.6.5.orig/CMakeLists.txt dotherside-0.6.5/CMakeLists.txt
--- dotherside-0.6.5.orig/CMakeLists.txt	2019-11-27 09:20:43.000000000 +0100
+++ dotherside-0.6.5/CMakeLists.txt	2020-05-19 21:27:38.281755940 +0200
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.0)
 
+option(ENABLE_DOCS "Enable docs" ON)
+option(ENABLE_TESTS "Enable tests" ON)
+option(ENABLE_STATIC_LIBS "Enable static libraries" ON)
+
 # Add additional source path for cmake
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/)
 
@@ -18,6 +22,13 @@
     endif()
 endif()
 
-add_subdirectory(doc)
 add_subdirectory(lib)
-add_subdirectory(test)
+
+if(ENABLE_DOCS)
+  add_subdirectory(doc)
+endif()
+
+if(ENABLE_TESTS)
+  add_subdirectory(test)
+endif()
+
diff -urN dotherside-0.6.5.orig/DOtherSide.pc.cmake dotherside-0.6.5/DOtherSide.pc.cmake
--- dotherside-0.6.5.orig/DOtherSide.pc.cmake	1970-01-01 01:00:00.000000000 +0100
+++ dotherside-0.6.5/DOtherSide.pc.cmake	2020-05-19 21:27:38.282755904 +0200
@@ -0,0 +1,14 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+sharedlibdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+
+Name: @PROJECT_NAME@
+Description: C language library for creating bindings for the Qt QML language
+Version: @DOTHERSIDE_VERSION@
+
+Requires: @PC_REQUIRES@
+Libs: -L${libdir} -lDOtherSide
+Cflags: -I${includedir}
+
diff -urN dotherside-0.6.5.orig/.gitignore dotherside-0.6.5/.gitignore
--- dotherside-0.6.5.orig/.gitignore	2019-11-27 09:20:43.000000000 +0100
+++ dotherside-0.6.5/.gitignore	2020-05-19 21:27:38.282755904 +0200
@@ -10,3 +10,29 @@
 .idea
 *.orig
 doc
+
+# libraries
+*.a
+*.so
+*.so.*
+
+# CMake
+CMakeLists.txt.user
+CMakeCache.txt
+CMakeFiles
+CMakeScripts
+Testing
+Makefile
+cmake_install.cmake
+install_manifest.txt
+compile_commands.json
+CTestTestfile.cmake
+_deps
+*_autogen
+CMakeDoxyfile.in
+CMakeDoxygenDefaults.cmake
+DOtherSide.pc
+
+# binaries
+test/TestDynamicQObject
+
diff -urN dotherside-0.6.5.orig/lib/CMakeLists.txt dotherside-0.6.5/lib/CMakeLists.txt
--- dotherside-0.6.5.orig/lib/CMakeLists.txt	2019-11-27 09:20:43.000000000 +0100
+++ dotherside-0.6.5/lib/CMakeLists.txt	2020-05-19 21:36:25.738085774 +0200
@@ -1,5 +1,7 @@
 project(DOtherSide)
 
+include(GNUInstallDirs)
+
 # Macro for merging common code between static and shared
 macro(add_target name type)
     find_package(Qt5 COMPONENTS Core Qml Gui Quick QuickControls2 Widgets)
@@ -38,8 +40,11 @@
 
     target_link_libraries(${name} PRIVATE Qt5::Core Qt5::Gui Qt5::Widgets Qt5::Qml Qt5::Quick)
 
+    # for DOtherSide.pc
+    set(PC_REQUIRES "Qt5Core, Qt5Gui, Qt5Widgets, Qt5Qml, Qt5Quick")
     if (${Qt5QuickControls2_FOUND})
         target_link_libraries(${name} PRIVATE Qt5::QuickControls2)
+        set(PC_REQUIRES "${PC_REQUIRES}, Qt5QuickControls2")
     endif()
 endmacro()
 
@@ -48,14 +53,24 @@
 set(major 0)
 set(minor 6)
 set(patch 4)
+set(DOTHERSIDE_VERSION "${major}.${minor}.${patch}")
+
 set_target_properties(${PROJECT_NAME}
 PROPERTIES
     SOVERSION "${major}.${minor}"
     VERSION "${major}.${minor}.${patch}"
 )
+install(TARGETS ${PROJECT_NAME}
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 
 # Add static version
-add_target(${PROJECT_NAME}Static STATIC)
+if(ENABLE_STATIC_LIBS)
+	add_target(${PROJECT_NAME}Static STATIC)
+	install(TARGETS ${PROJECT_NAME}Static
+		ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	)
+endif(ENABLE_STATIC_LIBS)
 
 # Install directive for header files
 install(FILES
@@ -64,8 +79,7 @@
   DESTINATION include/DOtherSide
 )
 
-# Install directive for binaries
-install(TARGETS ${PROJECT_NAME}
-  RUNTIME DESTINATION bin
-  LIBRARY DESTINATION lib
-)
+# pkg-config file
+configure_file(${CMAKE_SOURCE_DIR}/DOtherSide.pc.cmake ${CMAKE_BINARY_DIR}/DOtherSide.pc @ONLY)
+install(FILES ${CMAKE_BINARY_DIR}/DOtherSide.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
