diff -ur DOtherSide-483c1848e20f056649afa3e155c84a6d0cbcf907.orig/test/test_dotherside.cpp DOtherSide-483c1848e20f056649afa3e155c84a6d0cbcf907/test/test_dotherside.cpp
--- DOtherSide-483c1848e20f056649afa3e155c84a6d0cbcf907.orig/test/test_dotherside.cpp	2018-11-28 23:20:39.000000000 +0100
+++ DOtherSide-483c1848e20f056649afa3e155c84a6d0cbcf907/test/test_dotherside.cpp	2019-01-21 19:39:43.989006554 +0100
@@ -270,6 +270,7 @@
 
     void testLoadUrl()
     {
+        QSKIP("this one wants GLX");
         void *url = dos_qurl_create("qrc:///main.qml", QUrl::TolerantMode);
         QVERIFY(url != nullptr);
         dos_qqmlapplicationengine_load_url(m_engine, url);
@@ -281,6 +282,7 @@
 
     void testLoadData()
     {
+        QSKIP("this one wants GLX");
         dos_qqmlapplicationengine_load_data(m_engine, "import QtQuick 2.3; import QtQuick.Controls 1.2; ApplicationWindow { objectName: \"testWindow\"}");
         QCOMPARE(engine()->rootObjects().size(), 1);
         QCOMPARE(engine()->rootObjects().front()->objectName(), QString::fromLocal8Bit("testWindow"));
@@ -715,12 +717,14 @@
 private slots:
     void testCreate()
     {
+        QSKIP("this one wants GLX");
         VoidPointer view(dos_qquickview_create(), &dos_qquickview_delete);
         QVERIFY(view.get());
     }
 
     void testSourceAndSetSource()
     {
+        QSKIP("this one wants GLX");
         std::string testUrl = "qrc:/testQQuickView.qml";
         VoidPointer view(dos_qquickview_create(), &dos_qquickview_delete);
         VoidPointer url(dos_qurl_create(testUrl.c_str(), QUrl::StrictMode), &dos_qurl_delete);
