--- ext/imap/config.m4	2006-10-21 21:25:24.000000000 +0200
+++ ext/imap/config.m4	2006-10-21 21:26:15.000000000 +0200
@@ -192,12 +192,6 @@
       AC_MSG_ERROR(Cannot find rfc822.h. Please check your c-client installation.)
     fi
 
-    if test -r "$IMAP_DIR/c-client/c-client.a"; then
-      ln -s "$IMAP_DIR/c-client/c-client.a" "$IMAP_DIR/c-client/libc-client.a" >/dev/null 2>&1
-    elif test -r "$IMAP_DIR/$PHP_LIBDIR/c-client.a"; then
-      ln -s "$IMAP_DIR/$PHP_LIBDIR/c-client.a" "$IMAP_DIR/$PHP_LIBDIR/libc-client.a" >/dev/null 2>&1
-    fi
-
     for lib in c-client4 c-client imap; do
       IMAP_LIB=$lib
       IMAP_LIB_CHK($PHP_LIBDIR)
