diff -ur conky-1.20.0.orig/src/mouse-events.cc conky-1.20.0/src/mouse-events.cc
--- conky-1.20.0.orig/src/mouse-events.cc	2024-04-17 04:04:14.000000000 +0200
+++ conky-1.20.0/src/mouse-events.cc	2024-04-18 11:30:11.775782696 +0200
@@ -316,7 +316,7 @@
       "ConkyValuatorScrollMode",
   };
 
-  Atom override_atom = XInternAtom(display, atom_names[valuator >> 1], True);
+  Atom override_atom = XInternAtom(display, atom_names[valuator >> 1], False);
   unsigned char *value_return;
   Atom type_return;
   int format_return;
@@ -630,4 +630,4 @@
 }
 #endif /* BUILD_XINPUT */
 
-}  // namespace conky
\ No newline at end of file
+}  // namespace conky
