diff -ur conky-1.11.3.orig/cmake/ConkyPlatformChecks.cmake conky-1.11.3/cmake/ConkyPlatformChecks.cmake
--- conky-1.11.3.orig/cmake/ConkyPlatformChecks.cmake	2019-02-23 03:12:15.000000000 +0100
+++ conky-1.11.3/cmake/ConkyPlatformChecks.cmake	2019-06-15 16:43:58.654828121 +0200
@@ -213,12 +213,12 @@
 endif(BUILD_HTTP)
 
 if(BUILD_NCURSES)
-  include(FindCurses)
-  if(NOT CURSES_FOUND)
+	pkg_check_modules(NCURSES ncurses)
+	if(NOT NCURSES_FOUND)
     message(FATAL_ERROR "Unable to find ncurses library")
-  endif(NOT CURSES_FOUND)
-  set(conky_libs ${conky_libs} ${CURSES_LIBRARIES})
-  set(conky_includes ${conky_includes} ${CURSES_INCLUDE_DIR})
+	endif(NOT NCURSES_FOUND)
+	set(conky_libs ${conky_libs} ${NCURSES_LIBRARIES})
+	set(conky_includes ${conky_includes} ${NCURSES_INCLUDE_DIRS})
 endif(BUILD_NCURSES)
 
 if(BUILD_MYSQL)
