diff -ur conky-1.11.3.orig/src/CMakeLists.txt conky-1.11.3/src/CMakeLists.txt
--- conky-1.11.3.orig/src/CMakeLists.txt	2019-06-15 16:56:17.374337744 +0200
+++ conky-1.11.3/src/CMakeLists.txt	2019-06-15 16:58:11.081262475 +0200
@@ -210,6 +210,11 @@
   set(conky_libs ${conky_libs} tcp-portmon)
   set(port_monitors tcp-portmon.cc tcp-portmon.h)
   set(optional_sources ${optional_sources} ${port_monitors})
+  install(TARGETS
+  tcp-portmon
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+  ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 endif(BUILD_PORT_MONITORS)
 
 if(BUILD_X11)
@@ -300,7 +305,7 @@
 # Install libtcp-portmon too?
 install(TARGETS conky
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib
-        ARCHIVE DESTINATION lib)
+        LIBRARY DESTINATION ${LIB_INSTALL_DIR}
+        ARCHIVE DESTINATION ${LIB_INSTALL_DIR})
 
 print_target_properties(conky)
