diff -ur GTS-2.4.0.orig/Makefile.am GTS-2.4.0/Makefile.am
--- GTS-2.4.0.orig/Makefile.am	2017-07-06 07:00:00.000000000 +0200
+++ GTS-2.4.0/Makefile.am	2018-05-16 15:36:06.744981394 +0200
@@ -2,8 +2,7 @@
 CPPFLAGS += -DPKGDATADIR='"$(pkgdatadir)"'
 
 pkgdata_DATA = \
-			   sources/main/_gts-scan_area.txt \
-			   sources/main/gts_install_setup.txt
+			   sources/main/gts_initial_configuration.txt
 
 noinst_LTLIBRARIES = libgts.la
 
