diff -ur rpcs3-9999.orig/3rdparty/CMakeLists.txt rpcs3-9999/3rdparty/CMakeLists.txt
--- rpcs3-9999.orig/3rdparty/CMakeLists.txt	2023-05-04 17:31:43.171878938 +0200
+++ rpcs3-9999/3rdparty/CMakeLists.txt	2023-05-04 17:32:21.191571095 +0200
@@ -345,6 +345,8 @@
 			message(STATUS "Using system SDL2")
 			add_library(3rdparty_sdl2 INTERFACE)
 			target_compile_definitions(3rdparty_sdl2 INTERFACE -DHAVE_SDL2=1)
+			target_include_directories(3rdparty_sdl2 INTERFACE ${SDL2_INCLUDE_DIRS})
+			target_link_libraries(3rdparty_sdl2 INTERFACE ${SDL2_LIBRARIES})
 			set(SDL2_TARGET 3rdparty_sdl2)
 		else()
 			message(FATAL_ERROR "SDL2 is not available on this system")
