diff -urNp oiio-Release-1.8.13-orig/src/cmake/modules/FindOpenEXR.cmake oiio-Release-1.8.13-dwok/src/cmake/modules/FindOpenEXR.cmake
--- oiio-Release-1.8.13-orig/src/cmake/modules/FindOpenEXR.cmake	2018-10-17 22:17:04.717399785 +0200
+++ oiio-Release-1.8.13-dwok/src/cmake/modules/FindOpenEXR.cmake	2018-10-17 22:15:31.396193212 +0200
@@ -45,21 +45,21 @@ set (GENERIC_INCLUDE_PATHS
 # only the custom paths, then if that fails, try the default paths only.
 # This seems to be the most robust way I can find to not get confused when
 # both system and custom libraries are present.
-find_path (ILMBASE_INCLUDE_PATH OpenEXR/IlmBaseConfig.h
+find_path (ILMBASE_INCLUDE_PATH x86_64-pc-linux-gnu/OpenEXR/IlmBaseConfig.h
            PATHS ${GENERIC_INCLUDE_PATHS} NO_DEFAULT_PATH)
-find_path (ILMBASE_INCLUDE_PATH OpenEXR/IlmBaseConfig.h)
-find_path (OPENEXR_INCLUDE_PATH OpenEXR/OpenEXRConfig.h
+find_path (ILMBASE_INCLUDE_PATH x86_64-pc-linux-gnu/OpenEXR/IlmBaseConfig.h)
+find_path (OPENEXR_INCLUDE_PATH x86_64-pc-linux-gnu/OpenEXR/OpenEXRConfig.h
            PATHS ${GENERIC_INCLUDE_PATHS} NO_DEFAULT_PATH)
-find_path (OPENEXR_INCLUDE_PATH OpenEXR/OpenEXRConfig.h)
+find_path (OPENEXR_INCLUDE_PATH x86_64-pc-linux-gnu/OpenEXR/OpenEXRConfig.h)
 
 # Try to figure out version number
 if (EXISTS "${OPENEXR_INCLUDE_PATH}/OpenEXR/ImfMultiPartInputFile.h")
     # Must be at least 2.0
-    file(STRINGS "${OPENEXR_INCLUDE_PATH}/OpenEXR/OpenEXRConfig.h" TMP REGEX "^#define OPENEXR_VERSION_STRING .*$")
+    file(STRINGS "${OPENEXR_INCLUDE_PATH}/x86_64-pc-linux-gnu/OpenEXR/OpenEXRConfig.h" TMP REGEX "^#define OPENEXR_VERSION_STRING .*$")
     string (REGEX MATCHALL "[0-9]+[.0-9]+" OPENEXR_VERSION ${TMP})
-    file(STRINGS "${OPENEXR_INCLUDE_PATH}/OpenEXR/OpenEXRConfig.h" TMP REGEX "^#define OPENEXR_VERSION_MAJOR .*$")
+    file(STRINGS "${OPENEXR_INCLUDE_PATH}/x86_64-pc-linux-gnu/OpenEXR/OpenEXRConfig.h" TMP REGEX "^#define OPENEXR_VERSION_MAJOR .*$")
     string (REGEX MATCHALL "[0-9]+" OPENEXR_VERSION_MAJOR ${TMP})
-    file(STRINGS "${OPENEXR_INCLUDE_PATH}/OpenEXR/OpenEXRConfig.h" TMP REGEX "^#define OPENEXR_VERSION_MINOR .*$")
+    file(STRINGS "${OPENEXR_INCLUDE_PATH}/x86_64-pc-linux-gnu/OpenEXR/OpenEXRConfig.h" TMP REGEX "^#define OPENEXR_VERSION_MINOR .*$")
     string (REGEX MATCHALL "[0-9]+" OPENEXR_VERSION_MINOR ${TMP})
 else ()
     # Assume an old one, predates 2.x that had versions
