diff -ur DisplayCAL-3.9.11.orig/DisplayCAL/setup.py DisplayCAL-3.9.11/DisplayCAL/setup.py
--- DisplayCAL-3.9.11.orig/DisplayCAL/setup.py	2023-06-05 20:11:06.415064565 +0200
+++ DisplayCAL-3.9.11/DisplayCAL/setup.py	2023-06-05 20:11:55.364700265 +0200
@@ -695,7 +695,7 @@
             data_files.append(
                 (
                     os.path.join(os.path.dirname(data), "metainfo"),
-                    [os.path.join(pydir, "..", "dist", appstream_id + ".appdata.xml")],
+                    [os.path.join(pydir, "..", "misc", appstream_id + ".appdata.xml")],
                 )
             )
             data_files.append(
diff -ur DisplayCAL-3.9.11.orig/setup.py DisplayCAL-3.9.11/setup.py
--- DisplayCAL-3.9.11.orig/setup.py	2023-06-05 20:11:06.415064565 +0200
+++ DisplayCAL-3.9.11/setup.py	2023-06-05 20:11:34.574854992 +0200
@@ -690,7 +690,6 @@
         tmpl_name = appstream_id + ".appdata.xml"
         replace_placeholders(
             Path(pydir, "misc", tmpl_name),
-            Path(pydir, "dist", tmpl_name),
             lastmod_time,
             {
                 "APPDATAPROVIDES": provides,
