diff -ur gmic-qt-v.211.orig/CMakeLists.txt gmic-qt-v.211/CMakeLists.txt
--- gmic-qt-v.211.orig/CMakeLists.txt	2017-10-12 23:35:10.339891840 +0200
+++ gmic-qt-v.211/CMakeLists.txt	2017-10-12 23:38:28.162322846 +0200
@@ -328,10 +328,9 @@
 
 if (${GMIC_QT_HOST} STREQUAL "gimp")
 
-    find_package(Gimp2 REQUIRED)
-    find_package(Cairo REQUIRED)
-    find_package(GTK2 REQUIRED)
-    include_directories(${GIMP2_INCLUDE_DIRS} ${CAIRO_INCLUDE_DIRS} ${GTK2_INCLUDE_DIRS})
+    exec_program(gimptool-2.0 ARGS --libs-noui OUTPUT_VARIABLE GIMP2_LIBRARIES)
+    exec_program(gimptool-2.0 ARGS --cflags-noui OUTPUT_VARIABLE GIMP2_INCLUDE_DIRS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GIMP2_INCLUDE_DIRS}")
 
     set (gmic_qt_SRCS ${gmic_qt_SRCS} src/host_gimp.cpp)
     add_definitions(-DGMIC_HOST=gimp_qt -DGIMP_DISABLE_DEPRECATED)
