From 017e716dccd36b03d524fc707c0ebad6da9fc8b1 Mon Sep 17 00:00:00 2001
From: Lily Foster <lily@lily.flowers>
Date: Fri, 17 Mar 2023 14:14:23 -0400
Subject: [PATCH] Fix Qt form paths in CMakeLists

---
 CMakeLists.txt | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3375b72..8090050 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -560,18 +560,18 @@ set(gmic_qt_SRCS
 )
 
 set (gmic_qt_FORMS
-  dialogsettings.ui
-  filtersview.ui
-  headlessprogressdialog.ui
-  inoutpanel.ui
-  languageselectionwidget.ui
-  mainwindow.ui
-  multilinetextparameterwidget.ui
-  progressinfowidget.ui
-  progressinfowindow.ui
-  SearchFieldWidget.ui
-  sourceswidget.ui
-  zoomlevelselector.ui
+  ui/dialogsettings.ui
+  ui/filtersview.ui
+  ui/headlessprogressdialog.ui
+  ui/inoutpanel.ui
+  ui/languageselectionwidget.ui
+  ui/mainwindow.ui
+  ui/multilinetextparameterwidget.ui
+  ui/progressinfowidget.ui
+  ui/progressinfowindow.ui
+  ui/SearchFieldWidget.ui
+  ui/sourceswidget.ui
+  ui/zoomlevelselector.ui
 )
 
 if(ENABLE_DYNAMIC_LINKING)
