diff -ur gmic-qt-v.3.1.6.orig/translations/filters/csv2ts.sh gmic-qt-v.3.1.6/translations/filters/csv2ts.sh
--- gmic-qt-v.3.1.6.orig/translations/filters/csv2ts.sh	2022-08-16 11:18:26.000000000 +0200
+++ gmic-qt-v.3.1.6/translations/filters/csv2ts.sh	2022-12-07 17:43:18.568998678 +0100
@@ -65,11 +65,11 @@
 {
   local name="$1"
   local text="$2"
-  text=${text//&/&amp;}
-  text=${text//\"/&quot;}
-  text=${text//\'/&apos;}
-  text=${text//</&lt;}
-  text=${text//>/&gt;}
+  text=${text//&/\&amp;}
+  text=${text//\"/\&quot;}
+  text=${text//\'/\&apos;}
+  text=${text//</\&lt;}
+  text=${text//>/\&gt;}
   text=${text# }
   echo "      <${name}>${text}</$name>"
 }
diff -ur gmic-qt-v.3.1.6.orig/translations/filters/ts2csv.sh gmic-qt-v.3.1.6/translations/filters/ts2csv.sh
--- gmic-qt-v.3.1.6.orig/translations/filters/ts2csv.sh	2022-08-16 11:18:26.000000000 +0200
+++ gmic-qt-v.3.1.6/translations/filters/ts2csv.sh	2022-12-07 17:43:46.858730280 +0100
@@ -44,7 +44,7 @@
 function html2ascii()
 {
   local text="$1"
-  text=${text//&amp;/&}
+  text=${text//&amp;/\&}
   text=${text//&lt;/<}
   text=${text//&gt;/>}
   text=${text//&quot;/\"}
