diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0d61ff..61497f9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,9 @@ endif()
 set (GMIC_PATH "../gmic/src" CACHE STRING "Define the path to the gmic headers")
 message("G'MIC path: " ${GMIC_PATH})
 
+option(ENABLE_DYNAMIC_LINKING "Dynamically link the binaries to the GMIC shared library" OFF)
+set (GMIC_LIB_PATH "${GMIC_PATH}/.." CACHE STRING "Define the path to the GMIC shared library")
+
 #
 # Look for G'MIC repository
 #
@@ -362,9 +365,22 @@ set (gmic_qt_SRCS
   src/Widgets/LanguageSelectionWidget.cpp
   src/Widgets/ProgressInfoWindow.cpp
 
-  ${GMIC_PATH}/gmic.cpp
 )
 
+if(ENABLE_DYNAMIC_LINKING)
+  set(CMAKE_SKIP_RPATH TRUE)
+  set(gmic_qt_LIBRARIES
+    ${gmic_qt_LIBRARIES}
+    "gmic"
+    )
+  link_directories(${GMIC_LIB_PATH})
+else(ENABLE_DYNAMIC_LINKING)
+  set(gmic_qt_SRCS
+    ${gmic_qt_SRCS}
+  ${GMIC_PATH}/gmic.cpp
+    )
+endif(ENABLE_DYNAMIC_LINKING)
+
 qt5_wrap_ui(gmic_qt_SRCS
   ui/dialogsettings.ui
   ui/filtersview.ui
