diff -ur a/usr/share/applications/upwork.desktop b/usr/share/applications/upwork.desktop
--- a/usr/share/applications/upwork.desktop	2018-01-23 20:42:23.000000000 +0100
+++ b/usr/share/applications/upwork.desktop	2018-02-08 17:01:32.381272381 +0100
@@ -2,8 +2,8 @@
 Encoding=UTF-8
 Name=Upwork
 Exec=upwork
-Icon=upwork.png
+Icon=upwork
 Terminal=false
 Type=Application
-Categories=Development;Project Management;GTK
+Categories=Development;X-Project Management;GTK
 StartupNotify=false
