diff -ur qhull-2019.1.orig/src/libqhull/io.c qhull-2019.1/src/libqhull/io.c
--- qhull-2019.1.orig/src/libqhull/io.c	2019-06-29 02:30:15.000000000 +0200
+++ qhull-2019.1/src/libqhull/io.c	2019-12-28 23:52:21.800630496 +0100
@@ -72,8 +72,8 @@
     qh_memstatistics(qh ferr);
     d_1= (int)sizeof(setT) + (qh hull_dim - 1) * SETelemsize;
     qh_fprintf(qh ferr, 8040, "\
-    size in bytes: merge %d ridge %d vertex %d facet %d\n\
-         normal %d ridge vertices %d facet vertices or neighbors %d\n",
+    size in bytes: merge %zd ridge %zd vertex %zd facet %zd\n\
+         normal %d ridge vertices %d facet vertices or neighbors %zd\n",
             (int)sizeof(mergeT), (int)sizeof(ridgeT),
             (int)sizeof(vertexT), (int)sizeof(facetT),
             qh normal_size, d_1, d_1 + SETelemsize);
diff -ur qhull-2019.1.orig/src/libqhull/stat.c qhull-2019.1/src/libqhull/stat.c
--- qhull-2019.1.orig/src/libqhull/stat.c	2019-06-29 02:30:15.000000000 +0200
+++ qhull-2019.1/src/libqhull/stat.c	2019-12-28 23:54:24.820336469 +0100
@@ -502,7 +502,7 @@
   qh_allstatH();
   qh_allstatI();
   if (qhstat next > (int)sizeof(qhstat id)) {
-    qh_fprintf_stderr(6184, "qhull internal error (qh_initstatistics): increase size of qhstat.id[].  qhstat.next %d should be <= sizeof(qhstat id) %d\n", 
+    qh_fprintf_stderr(6184, "qhull internal error (qh_initstatistics): increase size of qhstat.id[].  qhstat.next %d should be <= sizeof(qhstat id) %zd\n", 
           qhstat next, (int)sizeof(qhstat id));
 #if 0 /* for locating error, Znumridges should be duplicated */
     for(i=0; i < ZEND; i++) {
