diff -ur client_generic.orig/Client/Makefile.am client_generic/Client/Makefile.am
--- client_generic.orig/Client/Makefile.am	2018-07-14 23:10:29.000000000 +0200
+++ client_generic/Client/Makefile.am	2019-01-20 19:04:38.172449040 +0100
@@ -12,6 +12,7 @@
 AM_CPPFLAGS = \
 	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
 	-DPACKAGE_LOCALE_DIR=\""$(datadir)/locale"\" \
+	-DGL_GLEXT_PROTOTYPES \
 	-I $(top_srcdir) \
 	-iquote ../DisplayOutput/OpenGL \
 	-I ../Common \
