diff -ur mcomix-git-486f02eef164df451a72598ce5989a1b37b49c60.orig/mcomix/image_tools.py mcomix-git-486f02eef164df451a72598ce5989a1b37b49c60/mcomix/image_tools.py
--- mcomix-git-486f02eef164df451a72598ce5989a1b37b49c60.orig/mcomix/image_tools.py	2019-04-06 11:41:56.000000000 +0200
+++ mcomix-git-486f02eef164df451a72598ce5989a1b37b49c60/mcomix/image_tools.py	2019-05-10 21:50:09.446199559 +0200
@@ -671,11 +671,11 @@
         supported_formats_gdk = {}
         for format in gtk.gdk.pixbuf_get_formats():
             name = format['name'].upper()
-            assert name not in supported_formats_gdk
-            supported_formats_gdk[name] = (
-                format['mime_types'],
-                format['extensions'],
-            )
+            if name not in supported_formats_gdk:
+                supported_formats_gdk[name] = (
+                    format['mime_types'],
+                    format['extensions'],
+                )
 
         # Step 3: merge format collections
         supported_formats = {}
