diff -ur mcomix-1.2.1.orig/mcomix/image_tools.py mcomix-1.2.1/mcomix/image_tools.py
--- mcomix-1.2.1.orig/mcomix/image_tools.py	2016-02-12 19:51:58.000000000 +0100
+++ mcomix-1.2.1/mcomix/image_tools.py	2019-04-08 12:22:58.234654973 +0200
@@ -551,11 +551,11 @@
         supported_formats = {}
         for format in gtk.gdk.pixbuf_get_formats():
             name = format['name'].upper()
-            assert name not in supported_formats
-            supported_formats[name] = (
-                format['mime_types'],
-                format['extensions'],
-            )
+            if name not in supported_formats:
+                supported_formats[name] = (
+                    format['mime_types'],
+                    format['extensions'],
+                )
     return supported_formats
 
 # Set supported image extensions regexp from list of supported formats.
