diff -ur cpupower.orig/Makefile cpupower/Makefile
--- cpupower.orig/Makefile	2019-07-08 00:41:56.000000000 +0200
+++ cpupower/Makefile	2019-09-05 18:51:39.147461326 +0200
@@ -178,12 +178,9 @@
 
 # if DEBUG is enabled, then we do not strip or optimize
 ifeq ($(strip $(DEBUG)),true)
-	override CFLAGS += -O1 -g -DDEBUG
-	STRIPCMD = /bin/true -Since_we_are_debugging
-else
-	override CFLAGS += $(OPTIMIZATION) -fomit-frame-pointer
-	STRIPCMD = $(STRIP) -s --remove-section=.note --remove-section=.comment
+	override CFLAGS += -DDEBUG
 endif
+STRIPCMD = true
 
 
 # the actual make rules
