diff -ur onnx-1.14.1.orig/CMakeLists.txt onnx-1.14.1/CMakeLists.txt
--- onnx-1.14.1.orig/CMakeLists.txt	2023-08-27 15:28:51.046225377 +0200
+++ onnx-1.14.1/CMakeLists.txt	2023-08-27 15:29:53.845832862 +0200
@@ -56,9 +56,9 @@
 endif()
 
 # Required to use /std:c++17 or higher on Windows
-# For other platforms, set C++11 as standard for the whole project
+# For other platforms, set C++14 as standard for the whole project
 if(NOT MSVC)
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 14)
 else()
   string(APPEND CMAKE_CXX_FLAGS " /std:c++17")
 endif()
