diff -ur blurb-1.1.0.orig/blurb.py blurb-1.1.0/blurb.py
--- blurb-1.1.0.orig/blurb.py	1990-08-09 16:25:15.000000000 +0200
+++ blurb-1.1.0/blurb.py	2022-07-23 16:39:00.074638705 +0200
@@ -701,7 +701,7 @@
     return process.stdout.decode('ascii')
 
 
-readme_re = re.compile(r"This is Python version [23]\.\d").match
+readme_re = re.compile(r"This is (Py|Tau)thon version [23]\.\d").match
 
 def chdir_to_repo_root():
     global root
