diff -ur wxPython-4.2.0.orig/buildtools/config.py wxPython-4.2.0/buildtools/config.py
--- wxPython-4.2.0.orig/buildtools/config.py	2022-08-08 05:17:44.000000000 +0200
+++ wxPython-4.2.0/buildtools/config.py	2022-09-27 01:13:12.532442649 +0200
@@ -228,8 +228,10 @@
             self.libdirs = []
             self.libs = []
 
-            self.cflags = self.getWxConfigValue('--cflags').split()
-            self.cxxflags = self.getWxConfigValue('--cxxflags').split()
+            self.cflags = self.getWxConfigValue('--cflags').split() + os.environ.get('CFLAGS', '').split()
+            print(self.cflags)
+            self.cxxflags = self.getWxConfigValue('--cxxflags').split() + os.environ.get('CXXFLAGS', '').split()
+            print(self.cxxflags)
             if self.debug:
                 for lst in [self.cflags, self.cxxflags]:
                     lst.append('-ggdb')
@@ -240,7 +242,8 @@
 
             lflags = self.getWxConfigValue('--libs')
             self.MONOLITHIC = (lflags.find("_xrc") == -1)
-            self.lflags = lflags.split()
+            self.lflags = lflags.split() + os.environ.get('LDFLAGS', '').split()
+            print(self.lflags)
 
             self.WXBASENAME = self.getWxConfigValue('--basename')
             self.WXRELEASE  = self.getWxConfigValue('--release')
