diff -ur ansel-4.0.0_p20230609.orig/src/CMakeLists.txt ansel-4.0.0_p20230609/src/CMakeLists.txt
--- ansel-4.0.0_p20230609.orig/src/CMakeLists.txt	2023-06-11 12:14:16.589498756 +0200
+++ ansel-4.0.0_p20230609/src/CMakeLists.txt	2023-06-11 12:20:46.776768805 +0200
@@ -605,7 +605,7 @@
     set(DT_REQ_INSTRUCTIONS "-msse2")
   endif()
 
-  include(march-mtune)
+  set(MARCH "")
 
   #if(NOT BUILD_SSE2_CODEPATHS)
   #  set(MARCH "${MARCH} -mno-sse2 -D__DISABLE_SSE2__ -U__SSE2__ -D__DISABLE_SSE__ -U__SSE__")
diff -ur ansel-4.0.0_p20230609.orig/src/external/rawspeed/cmake/compiler-flags.cmake ansel-4.0.0_p20230609/src/external/rawspeed/cmake/compiler-flags.cmake
--- ansel-4.0.0_p20230609.orig/src/external/rawspeed/cmake/compiler-flags.cmake	2023-06-11 12:14:16.259501065 +0200
+++ ansel-4.0.0_p20230609/src/external/rawspeed/cmake/compiler-flags.cmake	2023-06-11 12:21:19.196541952 +0200
@@ -1,5 +1,4 @@
 include(CheckCXXCompilerFlag)
-include(CpuMarch)
 include(CheckCXXCompilerFlagAndEnableIt)
 
 # yes, need to keep both the CMAKE_CXX_FLAGS and CMAKE_CXX_STANDARD.
