diff -ur ansel-4.0.0_pre20231120.orig/src/CMakeLists.txt ansel-4.0.0_pre20231120/src/CMakeLists.txt
--- ansel-4.0.0_pre20231120.orig/src/CMakeLists.txt	2023-11-24 10:47:56.494418458 +0100
+++ ansel-4.0.0_pre20231120/src/CMakeLists.txt	2023-11-24 10:49:39.633744205 +0100
@@ -598,7 +598,7 @@
     set(DT_REQ_INSTRUCTIONS "-msse2")
   endif()
 
-  include(march-mtune)
+  set(MARCH "")
 
   #if(NOT BUILD_SSE2_CODEPATHS)
   #  set(MARCH "${MARCH} -mno-sse2 -D__DISABLE_SSE2__ -U__SSE2__ -D__DISABLE_SSE__ -U__SSE__")
diff -ur ansel-4.0.0_pre20231120.orig/src/external/rawspeed/cmake/compiler-flags.cmake ansel-4.0.0_pre20231120/src/external/rawspeed/cmake/compiler-flags.cmake
--- ansel-4.0.0_pre20231120.orig/src/external/rawspeed/cmake/compiler-flags.cmake	2023-11-24 10:47:56.184420486 +0100
+++ ansel-4.0.0_pre20231120/src/external/rawspeed/cmake/compiler-flags.cmake	2023-11-24 10:50:14.773514486 +0100
@@ -1,4 +1,3 @@
-include(CpuMarch)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
