diff -ur tauthon-2.8.4.orig/setup.py tauthon-2.8.4/setup.py
--- tauthon-2.8.4.orig/setup.py	2023-01-29 17:33:34.962886875 +0100
+++ tauthon-2.8.4/setup.py	2023-01-29 17:35:30.251974658 +0100
@@ -2181,7 +2181,7 @@
                     ffi_lib = lib_name
                     break
 
-        if ffi_inc and ffi_lib:
+        if ffi_inc:
             ffi_headers = glob(os.path.join(ffi_inc, '*.h'))
             if grep_headers_for('ffi_prep_cif_var', ffi_headers):
                 ext.extra_compile_args.append("-DHAVE_FFI_PREP_CIF_VAR=1")
@@ -2191,8 +2191,9 @@
                 ext.extra_compile_args.append("-DHAVE_FFI_CLOSURE_ALLOC=1")
 
             ext.include_dirs.append(ffi_inc)
+
+        if ffi_lib:
             ext.libraries.append(ffi_lib)
-            self.use_system_libffi = True
 
         if sysconfig.get_config_var('HAVE_LIBDL'):
             # for dlopen, see bpo-32647
