diff -ur a/makefile.in b/makefile.in
--- a/makefile.in	2012-04-27 00:38:01.000000000 +0200
+++ b/makefile.in	2015-12-08 01:59:35.385883574 +0100
@@ -1068,11 +1068,10 @@
 
 install:
 	@echo "installing..."
-	mkdir -p $(exec_prefix)/lib
-	install -m 644 -p $(SHARED_LIB) libliquid.a $(exec_prefix)/lib
-	mkdir -p $(prefix)/include
-	mkdir -p $(prefix)/include/liquid
-	install -m 644 -p $(addprefix include/,$(headers_install)) $(prefix)/include/liquid
+	mkdir -p $(DESTDIR)@libdir@
+	mkdir -p $(DESTDIR)$(prefix)/include/liquid
+	install -m 644 -p $(SHARED_LIB) libliquid.a $(DESTDIR)@libdir@
+	install -m 644 -p $(addprefix include/,$(headers_install)) $(DESTDIR)$(prefix)/include/liquid
 	@echo ""
 	@echo "---------------------------------------------------------"
 	@echo "  liquid-dsp was successfully installed.     "
@@ -1081,7 +1080,7 @@
 	@echo "  libraries by running 'ldconfig' to make the shared"
 	@echo "  object available.  You might also need to modify your"
 	@echo "  LD_LIBRARY_PATH environment variable to include the"
-	@echo "  directory $(exec_prefix)"
+	@echo "  directory $(DESTDIR)@libdir@"
 	@echo ""
 	@echo "  Please report bugs to $(BUGREPORT)"
 	@echo "---------------------------------------------------------"
