diff -ur a/makefile.in b/makefile.in
--- a/makefile.in	2015-12-08 02:03:26.393096374 +0100
+++ b/makefile.in	2015-12-08 02:06:37.933639586 +0100
@@ -68,7 +68,7 @@
 INCLUDE_CFLAGS	= $(addprefix -I ,$(include_dirs))
 CONFIG_CFLAGS	= @ARCH_OPTION@ @PROFILE_OPTION@ @DEBUG_OPTION@ @SIMD_FLAGS@
 # -g : debugging info
-CFLAGS		+= $(INCLUDE_CFLAGS) -g -O2 -Wall -fPIC $(CONFIG_CFLAGS)
+CFLAGS		+= $(INCLUDE_CFLAGS) -Wall -fPIC $(CONFIG_CFLAGS)
 LDFLAGS		+= @LIBS@
 ARFLAGS		= r
 PATHSEP		= /
Only in a: makefile.in.rej
