diff -ur lmms.orig/plugins/carlabase/carla.h lmms/plugins/carlabase/carla.h
--- lmms.orig/plugins/carlabase/carla.h	2018-11-03 02:43:42.000000000 +0100
+++ lmms/plugins/carlabase/carla.h	2023-05-10 17:49:10.166315423 +0200
@@ -28,6 +28,7 @@
 #include <QtCore/QMutex>
 
 #define REAL_BUILD // FIXME this shouldn't be needed
+#include "CarlaDefines.h"
 #if CARLA_VERSION_HEX >= 0x010911
     #include "CarlaNativePlugin.h"
 #else
