diff -ur pytorch-v2.0.0.orig/c10/CMakeLists.txt pytorch-v2.0.0/c10/CMakeLists.txt
--- pytorch-v2.0.0.orig/c10/CMakeLists.txt	2023-04-04 02:12:06.603905392 +0200
+++ pytorch-v2.0.0/c10/CMakeLists.txt	2023-04-04 02:17:45.701427102 +0200
@@ -65,7 +65,7 @@
 if(${USE_GLOG})
     target_link_libraries(c10 PUBLIC glog::glog)
 endif()
-target_link_libraries(c10 PRIVATE fmt::fmt-header-only)
+target_link_libraries(c10 PRIVATE fmt)
 
 find_package(Backtrace)
 if(Backtrace_FOUND)
diff -ur pytorch-v2.0.0.orig/caffe2/CMakeLists.txt pytorch-v2.0.0/caffe2/CMakeLists.txt
--- pytorch-v2.0.0.orig/caffe2/CMakeLists.txt	2023-04-04 02:12:06.483906269 +0200
+++ pytorch-v2.0.0/caffe2/CMakeLists.txt	2023-04-04 02:26:18.007683171 +0200
@@ -106,7 +106,7 @@
 # Note: the folders that are being commented out have not been properly
 # addressed yet.
 
-if(NOT MSVC AND USE_XNNPACK)
+if(FALSE)
   if(NOT TARGET fxdiv)
     set(FXDIV_BUILD_TESTS OFF CACHE BOOL "")
     set(FXDIV_BUILD_BENCHMARKS OFF CACHE BOOL "")
@@ -533,7 +533,6 @@
 if(NOT MSVC)
   set_source_files_properties(${TORCH_SRC_DIR}/csrc/jit/tensorexpr/llvm_jit.cpp PROPERTIES COMPILE_FLAGS -Wno-noexcept-type)
   # Force -Werror on several files
-  set_source_files_properties(${CMAKE_CURRENT_LIST_DIR}/../aten/src/ATen/native/mkldnn/Pooling.cpp PROPERTIES COMPILE_FLAGS "-Werror")
 endif()
 # Disable certain warnings for GCC-9.X
 if(CMAKE_COMPILER_IS_GNUCXX AND (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 9.0.0))
@@ -811,7 +810,6 @@
 torch_compile_options(torch_cpu)  # see cmake/public/utils.cmake
 if(HAS_WERROR_SIGN_COMPARE AND WERROR)
   # target_compile_options(torch_cpu PRIVATE "-Werror=sign-compare")
-  set_property(SOURCE ${ATen_CORE_SRCS} ${ATen_CPU_SRCS} APPEND PROPERTY COMPILE_OPTIONS "-Werror=sign-compare")
 endif()
 
 set_property(SOURCE ${ATen_CORE_SRCS} APPEND
@@ -975,7 +973,6 @@
 endif()
 
 if(NOT MSVC AND USE_XNNPACK)
-  TARGET_LINK_LIBRARIES(torch_cpu PRIVATE fxdiv)
 endif()
 
 # ==========================================================
@@ -1095,8 +1092,7 @@
 target_include_directories(torch_cpu PRIVATE
   ${TORCH_ROOT}/third_party/miniz-2.1.0)
 
-target_include_directories(torch_cpu PRIVATE
-  ${TORCH_ROOT}/third_party/kineto/libkineto/include)
+target_include_directories(torch_cpu PRIVATE /usr/include/kineto)
 
 if(USE_KINETO)
   target_include_directories(torch_cpu PRIVATE
diff -ur pytorch-v2.0.0.orig/cmake/Codegen.cmake pytorch-v2.0.0/cmake/Codegen.cmake
--- pytorch-v2.0.0.orig/cmake/Codegen.cmake	2023-04-03 21:46:00.000000000 +0200
+++ pytorch-v2.0.0/cmake/Codegen.cmake	2023-04-04 02:23:25.178946202 +0200
@@ -57,7 +57,7 @@
   if(MSVC)
     set(OPT_FLAG "/fp:strict ")
   else(MSVC)
-    set(OPT_FLAG "-O3 ")
+    set(OPT_FLAG " ")
     if("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
       set(OPT_FLAG " ")
     endif()
diff -ur pytorch-v2.0.0.orig/cmake/Dependencies.cmake pytorch-v2.0.0/cmake/Dependencies.cmake
--- pytorch-v2.0.0.orig/cmake/Dependencies.cmake	2023-04-03 21:46:00.000000000 +0200
+++ pytorch-v2.0.0/cmake/Dependencies.cmake	2023-04-04 02:28:00.806931915 +0200
@@ -486,7 +486,7 @@
 list(APPEND Caffe2_DEPENDENCY_LIBS cpuinfo)
 
 # ---[ QNNPACK
-if(USE_QNNPACK)
+if(FALSE)
   set(CAFFE2_THIRD_PARTY_ROOT "${PROJECT_SOURCE_DIR}/third_party")
 
   if(NOT DEFINED QNNPACK_SOURCE_DIR)
@@ -542,7 +542,7 @@
 endif()
 
 # ---[ Caffe2 Int8 operators (enabled by USE_QNNPACK) depend on gemmlowp and neon2sse headers
-if(USE_QNNPACK)
+if(FALSE)
   set(CAFFE2_THIRD_PARTY_ROOT "${PROJECT_SOURCE_DIR}/third_party")
   include_directories(SYSTEM "${CAFFE2_THIRD_PARTY_ROOT}/gemmlowp")
   include_directories(SYSTEM "${CAFFE2_THIRD_PARTY_ROOT}/neon2sse")
@@ -1551,7 +1551,6 @@
       set_target_properties(onnx_proto PROPERTIES CXX_STANDARD 17)
     endif()
   endif()
-  add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/../third_party/foxi EXCLUDE_FROM_ALL)
 
   add_definitions(-DONNX_NAMESPACE=${ONNX_NAMESPACE})
   if(NOT USE_SYSTEM_ONNX)
@@ -1819,7 +1818,6 @@
 #
 set(TEMP_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libs" FORCE)
-add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/fmt)
 
 # Disable compiler feature checks for `fmt`.
 #
@@ -1828,9 +1826,7 @@
 # CMAKE_CXX_FLAGS in ways that break feature checks. Since we already know
 # `fmt` is compatible with a superset of the compilers that PyTorch is, it
 # shouldn't be too bad to just disable the checks.
-set_target_properties(fmt-header-only PROPERTIES INTERFACE_COMPILE_FEATURES "")
 
-list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt-header-only)
 set(BUILD_SHARED_LIBS ${TEMP_BUILD_SHARED_LIBS} CACHE BOOL "Build shared libs" FORCE)
 
 # ---[ Kineto
diff -ur pytorch-v2.0.0.orig/cmake/External/nnpack.cmake pytorch-v2.0.0/cmake/External/nnpack.cmake
--- pytorch-v2.0.0.orig/cmake/External/nnpack.cmake	2023-04-03 21:46:00.000000000 +0200
+++ pytorch-v2.0.0/cmake/External/nnpack.cmake	2023-04-04 02:26:48.387461156 +0200
@@ -58,7 +58,7 @@
   set(PTHREADPOOL_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/pthreadpool" CACHE STRING "pthreadpool source directory")
   set(GOOGLETEST_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/googletest" CACHE STRING "Google Test source directory")
 
-  if(NOT TARGET nnpack)
+  if(FALSE)
     if(NOT USE_SYSTEM_PTHREADPOOL AND USE_INTERNAL_PTHREADPOOL_IMPL)
       set(NNPACK_CUSTOM_THREADPOOL ON CACHE BOOL "")
     endif()
diff -ur pytorch-v2.0.0.orig/cmake/public/utils.cmake pytorch-v2.0.0/cmake/public/utils.cmake
--- pytorch-v2.0.0.orig/cmake/public/utils.cmake	2023-04-03 21:46:00.000000000 +0200
+++ pytorch-v2.0.0/cmake/public/utils.cmake	2023-04-04 02:22:52.489185098 +0200
@@ -507,8 +507,6 @@
   endif()
 
   # Use -O2 for release builds (-O3 doesn't improve perf, and -Os results in perf regression)
-  target_compile_options(${libname} PRIVATE
-      $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<OR:$<CONFIG:Release>,$<CONFIG:RelWithDebInfo>>>:-O2>)
 
 endfunction()
 
diff -ur pytorch-v2.0.0.orig/CMakeLists.txt pytorch-v2.0.0/CMakeLists.txt
--- pytorch-v2.0.0.orig/CMakeLists.txt	2023-04-04 02:12:07.723897204 +0200
+++ pytorch-v2.0.0/CMakeLists.txt	2023-04-04 02:22:03.839540632 +0200
@@ -799,12 +799,11 @@
 
 # ---[ Build flags
 if(NOT MSVC)
-  string(APPEND CMAKE_CXX_FLAGS " -O2 -fPIC")
+  string(APPEND CMAKE_CXX_FLAGS " -fPIC")
   # Eigen fails to build with some versions, so convert this to a warning
   # Details at http://eigen.tuxfamily.org/bz/show_bug.cgi?id=1459
   string(APPEND CMAKE_CXX_FLAGS " -Wall")
   string(APPEND CMAKE_CXX_FLAGS " -Wextra")
-  append_cxx_flag_if_supported("-Werror=return-type" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Werror=non-virtual-dtor" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Werror=braced-scalar-init" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-Werror=range-loop-construct" CMAKE_CXX_FLAGS)
@@ -912,8 +911,6 @@
   string(APPEND CMAKE_LINKER_FLAGS_DEBUG " -fno-omit-frame-pointer -O0")
   append_cxx_flag_if_supported("-fno-math-errno" CMAKE_CXX_FLAGS)
   append_cxx_flag_if_supported("-fno-trapping-math" CMAKE_CXX_FLAGS)
-  append_cxx_flag_if_supported("-Werror=format" CMAKE_CXX_FLAGS)
-  append_cxx_flag_if_supported("-Werror=cast-function-type" CMAKE_CXX_FLAGS)
 endif()
 
 if(USE_ASAN)
diff -ur pytorch-v2.0.0.orig/torch/CMakeLists.txt pytorch-v2.0.0/torch/CMakeLists.txt
--- pytorch-v2.0.0.orig/torch/CMakeLists.txt	2023-04-04 02:12:06.593905465 +0200
+++ pytorch-v2.0.0/torch/CMakeLists.txt	2023-04-04 02:19:23.450712752 +0200
@@ -66,15 +66,9 @@
     ${CMAKE_BINARY_DIR}
     ${CMAKE_BINARY_DIR}/aten/src
     ${CMAKE_BINARY_DIR}/caffe2/aten/src
-    ${CMAKE_BINARY_DIR}/third_party
-    ${CMAKE_BINARY_DIR}/third_party/onnx
 
-    ${TORCH_ROOT}/third_party/valgrind-headers
 
-    ${TORCH_ROOT}/third_party/gloo
-    ${TORCH_ROOT}/third_party/onnx
-    ${TORCH_ROOT}/third_party/flatbuffers/include
-    ${TORCH_ROOT}/third_party/kineto/libkineto/include
+    /usr/include/kineto
 
     ${TORCH_SRC_DIR}/csrc
     ${TORCH_SRC_DIR}/csrc/api/include
@@ -87,7 +81,6 @@
     python::python
     pybind::pybind11
     shm
-    fmt::fmt-header-only
     ATEN_CPU_FILES_GEN_LIB)
 
 set(TORCH_PYTHON_COMPILE_DEFINITIONS)
