diff '--color=auto' -ur cpuid-20241023.orig/Makefile cpuid-20241023/Makefile
--- cpuid-20241023.orig/Makefile	2024-10-23 15:37:40.000000000 +0200
+++ cpuid-20241023/Makefile	2024-11-04 19:00:00.189907545 +0100
@@ -1,4 +1,3 @@
-CFLAGS+=-g
 CPPFLAGS?=
 LDFLAGS?=
 ifneq (,$(findstring arch=i386,$(CFLAGS)))
@@ -6,7 +5,6 @@
 endif
 CFL=$(CPPFLAGS) $(CFLAGS) $(CISA) -Wall -W -Wshadow -Wcast-align -Wredundant-decls -Wbad-function-cast -Wcast-qual -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wimplicit-fallthrough -Wunused-parameter -Wundef -D_FILE_OFFSET_BITS=64 -DVERSION=$(VERSION)
 CFL_OPT=-O3 $(CFL) -Wformat-overflow=2 
-INSTALL_STRIP=-s
 
 PACKAGE=cpuid
 VERSION=20241023
@@ -41,7 +39,7 @@
 
 BUILDROOT=$(DESTDIR)
 
-default: $(PROG) $(PROG).man.gz cpuinfo2cpuid cpuinfo2cpuid.man.gz
+default: $(PROG) cpuinfo2cpuid
 
 $(PROG): cpuid.c Makefile
 	$(CC) $(CFL) $(LDFLAGS) -o $@ cpuid.c
@@ -60,11 +58,11 @@
 cpuinfo2cpuid.man.gz: cpuinfo2cpuid.man
 	gzip < $< > $@
 
-install: $(PROG) $(PROG).man.gz cpuinfo2cpuid cpuinfo2cpuid.man.gz
+install: $(PROG) $(PROG).man.gz cpuinfo2cpuid cpuinfo2cpuid.man
 	install -D $(INSTALL_STRIP) -m 755 $(PROG) $(BUILDROOT)/usr/bin/$(PROG)
-	install -D -m 444 $(PROG).man.gz       $(BUILDROOT)/usr/share/man/man1/$(PROG).1.gz
+	install -D -m 444 $(PROG).man       $(BUILDROOT)/usr/share/man/man1/$(PROG).1
 	install -D -m 755 cpuinfo2cpuid        $(BUILDROOT)/usr/bin/cpuinfo2cpuid
-	install -D -m 444 cpuinfo2cpuid.man.gz $(BUILDROOT)/usr/share/man/man1/cpuinfo2cpuid.1.gz
+	install -D -m 444 cpuinfo2cpuid.man $(BUILDROOT)/usr/share/man/man1/cpuinfo2cpuid.1
 
 clean:
 	rm -f $(PROG) $(PROG).i386 $(PROG).x86_64
