diff --git a/perf/_system.py b/perf/_system.py
index f18d2fd..1c01bbe 100644
--- a/perf/_system.py
+++ b/perf/_system.py
@@ -143,7 +143,11 @@ class TurboBoostMSR(Operation):
         try:
             fd = os.open(path, os.O_RDONLY)
             try:
-                data = os.pread(fd, size, reg_num)
+                if hasattr(os, 'pread'):
+                    data = os.pread(fd, size, reg_num)
+                else:
+                    os.lseek(fd, reg_num, os.SEEK_SET)
+                    data = os.read(fd, size)
             finally:
                 os.close(fd)
         except OSError as exc:
@@ -211,7 +215,11 @@ class TurboBoostMSR(Operation):
         try:
             fd = os.open(path, os.O_WRONLY)
             try:
-                data = os.pwrite(fd, data, reg_num)
+                if hasattr(os, 'pwrite'):
+                    data = os.pwrite(fd, data, reg_num)
+                else:
+                    os.lseek(fd, reg_num, os.SEEK_SET)
+                    os.write(fd, data)
             finally:
                 os.close(fd)
         except IOError as exc:
