From 4e43d055206ca57794bdcaad2dac8978fd960663 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Tue, 31 Aug 2021 10:50:53 +0200
Subject: [PATCH 16/36] Fix accidentally leaving one sub-test commented out

---
 Lib/test/test_smtplib.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Lib/test/test_smtplib.py b/Lib/test/test_smtplib.py
index 2224524ebd..b6fa293228 100644
--- a/Lib/test/test_smtplib.py
+++ b/Lib/test/test_smtplib.py
@@ -286,7 +286,7 @@ class DebuggingServerTests(unittest.TestCase):
             timeout=10)  # support.LOOPBACK_TIMEOUT in newer Pythons
 
         self.addCleanup(smtp.close)
-#        smtp.sendmail("hi@me.com", "you@me.com", m)
+        smtp.sendmail("hi@me.com", "you@me.com", m)
         # NB: gross hack but still cleaner than backporting whole ESMTP
         # support to DebuggingServer
         smtp.does_esmtp = 1
-- 
2.38.1

