From 94de8848fcdd226ff3868ae073ff2751f4fcb5f7 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Sun, 17 Jun 2018 11:34:34 -0400
Subject: [PATCH 29/36] Disable nis

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 89946bcd55..c240376e07 100644
--- a/setup.py
+++ b/setup.py
@@ -1407,7 +1407,7 @@ class PyBuildExt(build_ext):
             else:
                 missing.append('resource')
 
-            nis = self._detect_nis(inc_dirs, lib_dirs)
+            nis = None
             if nis is not None:
                 exts.append(nis)
             else:
-- 
2.38.1

